yarl/__init__.py

```diff
@@ -745,7 +745,7 @@ def _get_str_query(self, *args, **kwargs):
             raise ValueError("Either kwargs or single query parameter "
                              "must be present")
 
-        if query is None:
+        if not query:
```

Comment:
Please no.
Your change allows to pass `url.with_query(0)` or `url.with_query(False)`.
Better to add a check for empty mapping 3 lines below.

Reply:
Fair point, pushing a change.